test/teiserver_web/controllers/account/security_controller_test.exs

```diff
@@ -13,4 +15,78 @@ defmodule TeiserverWeb.Account.SecurityControllerTest do
     conn = GeneralTestLib.login(conn, user.email)
     assert redirected_to(conn) == ~p"/teiserver/account/security/edit_password"
   end
+
+  describe "OAuth application revocation" do
+    setup do
+      {:ok, kw} =
+        GeneralTestLib.conn_setup(Teiserver.TeiserverTestLib.player_permissions())
+        |> Teiserver.TeiserverTestLib.conn_setup()
+
+      {:ok, conn} = Keyword.fetch(kw, :conn)
+      {:ok, user} = Keyword.fetch(kw, :user)
+
+      # Create an OAuth application
+      {:ok, app} =
+        OAuth.create_application(%{
+          name: "Test App",
+          uid: "test_app",
+          owner_id: user.id,
+          scopes: ["tachyon.lobby"],
+          redirect_uris: ["http://localhost/callback"]
+        })
+
+      # Create tokens and codes for the app
+      token = OAuthFixtures.token_attrs(user.id, app) |> OAuthFixtures.create_token()
+      code = OAuthFixtures.code_attrs(user.id, app) |> OAuthFixtures.create_code()
+
+      {:ok, conn: conn, user: user, app: app, token: token, code: code}
+    end
+
+    test "successfully revokes OAuth application access", %{
+      conn: conn,
+      app: app,
+      token: token,
+      code: code
+    } do
+      # Verify tokens and codes exist
+      assert Teiserver.OAuth.TokenQueries.get_token(token.value)
+      assert Teiserver.OAuth.CodeQueries.get_code(code.value)
+
+      # Revoke access
+      conn = delete(conn, ~p"/teiserver/account/security/revoke_oauth/#{app.id}")
+
+      # Should redirect back to security page
+      assert redirected_to(conn) == ~p"/teiserver/account/security"
+
+      # Should show success message
+      assert Phoenix.Flash.get(conn.assigns.flash, :info) ==
+               "OAuth application access revoked successfully."
```

Comment:
I'm not a fan of checking for specific human readable output. A simple copy change would mean you also need to change the test, this brings no value.
You could check that there is a flash message and call it a day.